ako-gateway-api/status/gatewayclass_status.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023-2024 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	akogatewayapilib "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/status"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+)
+
+type gatewayClass struct{}
+
+func (o *gatewayClass) Get(key string, name string) *gatewayv1beta1.GatewayClass {
+
+	obj, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().GatewayClassInformer.Lister().Get(name)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: unable to get the GatewayClass object. err: %s", key, err)
+		return nil
+	}
+	utils.AviLog.Debugf("key: %s, msg: Successfully retrieved the GatewayClass object %s", key, name)
+	return obj.DeepCopy()
+}
+
+func (o *gatewayClass) GetAll(key string) map[string]*gatewayv1beta1.GatewayClass {
+
+	objs, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().GatewayClassInformer.Lister().List(labels.Everything())
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: unable to get the GatewayClass objects. err: %s", key, err)
+		return nil
+	}
+
+	gatewayClassMap := make(map[string]*gatewayv1beta1.GatewayClass)
+	for _, obj := range objs {
+		gatewayClassMap[obj.Name] = obj.DeepCopy()
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: Successfully retrieved the GatewayClass objects", key)
+	return gatewayClassMap
+}
+
+func (o *gatewayClass) Delete(key string, option status.StatusOptions) {
```

Comment:
What is plan of handling these functions?

Reply:
GatewayClass status is set during validation itself. so rest layer won't update gateway class, these are likely to remain placeholder. we will revisit there when we add support for parameterRef (CRDs(